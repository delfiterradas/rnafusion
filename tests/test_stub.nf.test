nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    profile "test"
    options "-stub" // TODO remove once reference files are available
    tag "pipeline"
    tag "pipeline_rnafusion"
    tag "test_stub"

    test("stub test with fastp trim") {

        when {
            params {
                outdir          = "$outputDir"
                fastp_trim      = true
                fusion_annot_lib           = 'https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/master/CTAT_HumanFusionLib.mini.dat.gz'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // Ignore files with timestamps in their names
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            // def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore') //TODO uncomment once -stub is removed
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnafusion_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    // stable_path // TODO uncomment once -stub is removed
                ).match() }
            )
        }
    }

    test("stub test no fastp trim") {

        when {
            params {
                outdir          = "$outputDir"
                fusion_annot_lib           = 'https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/master/CTAT_HumanFusionLib.mini.dat.gz'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // Ignore files with timestamps in their names
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            // def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore') //TODO uncomment once -stub is removed
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnafusion_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    // stable_path // TODO uncomment once -stub is removed
                ).match() }
            )
        }
    }
}
