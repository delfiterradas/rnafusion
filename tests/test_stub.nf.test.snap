{
    "stub test starfusion": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T09:52:16.204798"
    },
    "stub test stringtie": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T09:52:44.54581"
    },
    "stub test arriba": {
        "content": [
            16,
            {
                "ARRIBA_ARRIBA": {
                    "arriba": "2.4.0"
                },
                "ARRIBA_DOWNLOAD": {
                    "arriba_download": "2.4.0"
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GATK4_BEDTOINTERVALLIST": {
                    "gatk4": "4.6.1.0"
                },
                "GATK4_CREATESEQUENCEDICTIONARY": {
                    "gatk4": "4.6.1.0"
                },
                "GATK4_MARKDUPLICATES": {
                    "gatk4": "4.6.1.0",
                    "samtools": 1.21
                },
                "GENCODE_DOWNLOAD": {
                    "wget": "1.21.4"
                },
                "GET_RRNA_TRANSCRIPTS": {
                    "get_rrna_transcripts": "v1.0"
                },
                "GTF_TO_REFFLAT": {
                    "gtfToGenePred": 377
                },
                "PICARD_COLLECTINSERTSIZEMETRICS": {
                    "picard": "3.3.0"
                },
                "PICARD_COLLECTRNASEQMETRICS": {
                    "picard": "3.3.0"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_INDEX": {
                    "samtools": 1.21
                },
                "STAR_ALIGN": {
                    "star": "2.7.11b",
                    "samtools": 1.21,
                    "gawk": "5.1.0"
                },
                "STAR_GENOMEGENERATE": {
                    "star": "2.7.11b",
                    "samtools": 1.21,
                    "gawk": "5.1.0"
                },
                "Workflow": {
                    "nf-core/rnafusion": "v4.0.0dev"
                }
            },
            [
                "arriba",
                "arriba/test.arriba.fusions.discarded.tsv",
                "arriba/test.arriba.fusions.tsv",
                "fastqc",
                "fastqc/test.html",
                "fastqc/test.zip",
                "gatk4",
                "gatk4/Homo_sapiens.GRCh38.46_dna_primary_assembly.dict",
                "get",
                "get/rrna.bed",
                "get/rrna.gtf",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_report.html",
                "picard",
                "picard/test.bai",
                "picard/test.bam",
                "picard/test.bam.metrics",
                "picard/test.cram",
                "picard/test.cram.crai",
                "picard/test.rna_metrics",
                "picard/test_collectinsertsize.pdf",
                "picard/test_collectinsertsize.txt",
                "pipeline_info",
                "pipeline_info/nf_core_rnafusion_software_mqc_versions.yml",
                "references",
                "references/arriba",
                "references/arriba/blacklist_hg38_GRCh38_v2.4.0.tsv.gz",
                "references/arriba/cytobands_hg38_GRCh38_v2.4.0.tsv",
                "references/arriba/known_fusions_hg38_GRCh38_v2.4.0.tsv.gz",
                "references/arriba/protein_domains_hg38_GRCh38_v2.4.0.gff3",
                "references/gencode",
                "references/gencode/Homo_sapiens.GRCh38.46.gtf",
                "references/gencode/Homo_sapiens.GRCh38.46.gtf.interval_list",
                "references/gencode/Homo_sapiens.GRCh38.46.gtf.refflat",
                "references/gencode/Homo_sapiens.GRCh38.46_dna_primary_assembly.fa",
                "references/gencode/Homo_sapiens.GRCh38.46_dna_primary_assembly.fa.fai",
                "references/star",
                "references/star/Genome",
                "references/star/Log.out",
                "references/star/SA",
                "references/star/SAindex",
                "references/star/chrLength.txt",
                "references/star/chrName.txt",
                "references/star/chrNameLength.txt",
                "references/star/chrStart.txt",
                "references/star/exonGeTrInfo.tab",
                "references/star/exonInfo.tab",
                "references/star/geneInfo.tab",
                "references/star/genomeParameters.txt",
                "references/star/sjdbInfo.txt",
                "references/star/sjdbList.fromGTF.out.tab",
                "references/star/sjdbList.out.tab",
                "references/star/transcriptInfo.tab",
                "star",
                "star/test.Aligned.sortedByCoord.out.bam",
                "star/test.Aligned.sortedByCoord.out.bam.bai",
                "star/test.Aligned.unsort.out.bam",
                "star/test.Chimeric.out.junction",
                "star/test.Log.final.out",
                "star/test.Log.out",
                "star/test.Log.progress.out",
                "star/test.ReadsPerGene.out.tab",
                "star/test.SJ.out.tab",
                "star/test.Signal.UniqueMultiple.str1.out.bg",
                "star/test.Signal.UniqueMultiple.str1.out.wig",
                "star/test.out.sam",
                "star/test.sortedByCoord.out.bam",
                "star/test.tab",
                "star/test.toTranscriptome.out.bam",
                "star/test.unmapped_1.fastq.gz",
                "star/test.unmapped_2.fastq.gz",
                "star/testXd.out.bam"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T14:02:08.078812972"
    },
    "stub test fusioninspector": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T09:54:26.426493"
    },
    "stub test salmon": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T09:54:00.321387"
    },
    "stub test ctatsplicing": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T09:50:52.1988"
    },
    "stub test fusionreport": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T09:53:09.814376"
    },
    "stub test fastp": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T09:53:35.550144"
    },
    "stub test fusioncatcher": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T09:51:23.745133"
    },
    "stub test all": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T09:21:44.763989"
    }
}